### ISSUE TITLE
RNN Transducer Loss

### ISSUE THREAD

**Issue by vincentqb** [pytorch/audio#1240]
This issue is to track the follow-up work to #1171, which introduced `rnnt_loss` and `RNNTLoss` as a [prototype](https://pytorch.org/audio/stable/index.html) in `torchaudio.prototype.transducer` using [HawkAaron's warp-transducer](https://github.com/HawkAaron/warp-transducer).

- Update documentation
  - [ ] Guard documentation (e.g. with [conditional doc build](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/conditions.html))
  - [x] Write e.g. [torchtext](https://pytorch.org/text/master/experimental_datasets.html) (#1171)
- Extend guards for prototype
  - [ ] Guard prototype python files by omitting them from torchaudio, see also https://github.com/pytorch/audio/pull/1137#discussion_r551496192
  - [x] Guard building third party transducer even if not added as an extension (#1159)
  - [ ] Enable building transducer in nightlies only, disable for release.
- Update building process
  - [ ] Pass along the `DEBUG` flag to cmake
  - [x] Remove hardcoded O2/O3 optimization, see https://github.com/pytorch/audio/pull/1137#discussion_r551498022 (#1159)
  - [x] Build within same folders as libsox, https://github.com/pytorch/audio/pull/1137#discussion_r551499829 and https://github.com/pytorch/audio/pull/1137#discussion_r551556305 (#1159)
  - [x] Move libsox to a third_party subfolder as suggested in https://github.com/pytorch/audio/pull/1137#discussion_r550321378 (#1161).
  - [ ] Add GPU implementation and compilation.
- Modernization
  - [ ] Migrate the [checks](https://github.com/pytorch/audio/pull/1137/files#diff-9623f973feeae59fa7215b73ad28d00f83c59bfa00f0743c096e4ab36fe5e25dR133) to C++.
  - [ ] Add Torchscriptability test ([internal](https://fb.workplace.com/groups/329222650990087/permalink/830958927483121/)).
  - [ ] Investigate using [`AT_DISPATCH_FLOATING_TYPES`](https://github.com/pytorch/pytorch/blob/22a34bcf4e5eaa348f0117c414c3dd760ec64b13/aten/src/ATen/native/cpu/Activation.cpp#L61).
  - [x] Update bindings to remove pytorch deprecation warnings. (#1160)
  - [ ] Refactor and update the API, see [warprnnt](https://github.com/1ytic/warp-rnnt) and [internal](https://fb.quip.com/tzrDA3DHqlY8).
  - [ ] Add support for `float16`.

cc @astaff, [internal](https://fb.quip.com/tzrDA3DHqlY8)

**Comment by maxwellzh** [comment]
Is there a plan to support the packed layout logits of RNNT loss?

Ref: Sec 3.1 https://arxiv.org/abs/1909.12415
